integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noark/NoarkClientFactory.java

```diff
@@ -11,30 +14,36 @@
  */
 public class NoarkClientFactory {
 
-    private static final String E_PHORTE = "ePhorte";
-    private static final String P360 = "P360";
-    private static final String WEBSAK = "WebSak";
+    private static final String E_PHORTE = "ephorte";
+    private static final String P360 = "p360";
+    private static final String WEBSAK = "websak";
+    private static final String MAIL = "mail";
     private final NoarkClientSettings settings;
 
     public NoarkClientFactory(NoarkClientSettings settings) {
         this.settings = settings;
     }
 
-    public NoarkClient from(IntegrasjonspunktProperties.NorskArkivstandardSystem properties) throws UnknownArchiveSystemException {
-        String noarkType = properties.getType();
-        if (noarkType == null) {
-            String message = "You need to configure what type of archive system you are using. Valid ones are %s, %s";
-            throw new MissingConfigurationException(String.format(message, E_PHORTE, P360));
+    public NoarkClient from(IntegrasjonspunktProperties properties) throws UnknownArchiveSystemException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "from" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)